worker/worker.py

```diff
@@ -1430,40 +1432,52 @@ def fetch_and_handle_task(
             req = send_api_post_request(api, payload)
         except Exception as e:
             print("Exception posting failed_task:\n", e, sep="", file=sys.stderr)
+
+    def upload_pgn_data(pgn_data, run_id, task_id, remote, payload):
+        with io.BytesIO() as gz_buffer:
+            with gzip.GzipFile(
+                filename=f"{str(run_id)}-{task_id}.pgn.gz",
+                mode="wb",
+                fileobj=gz_buffer,
+            ) as gz:
+                gz.write(pgn_data.encode())
+            payload["pgn"] = base64.b64encode(gz_buffer.getvalue()).decode()
+
+        print("Uploading compressed PGN of {} bytes".format(len(payload["pgn"])))
+        return send_api_post_request(remote + "/api/upload_pgn", payload)
+
+    if not pgn_file["name"] or not pgn_file["name"].exists():
```

Comment:
Better also check for zero-sized files, which otherwise leads to a crc check error:

```diff
-    if not pgn_file["name"] or not pgn_file["name"].exists():
+    if not pgn_file["name"] or not pgn_file["name"].exists() or pgn_file["name"].stat().st_size == 0:
```